### PR TITLE
Remove the systemd socket activation functionality

### PR DESCRIPTION
**Pull request by mike-sul**
Signed-off-by: Mykhaylo Sul <myk.sul@gmail.com>

### CODE REVIEW

**Comment by lbonn** [comment]
You should be able to remove `BUILD_SYSTEMD` and everything related.

**Comment by mike-sul** [comment]
> You should be able to remove `BUILD_SYSTEMD` and everything related.

Yeah, I hope so :)

File: thirdparty.spdx [hunk 1]
```diff
@@ -208,19 +208,6 @@ FilesAnalyzed: false
 PackageComment: <text>Dynamically linked.</text>
 
 
-PackageName: systemd
```
Comment (lbonn) [inline thirdparty.spdx]: There is related line, close to the top of this file:

```Relationship: SPDXRef-DOCUMENT DESCRIBES SPDXRef-systemd```

**Comment by mike-sul** [comment]
> Looks good, but I still see a reference to `socket_activation` in `docs/doxygen/Doxyfile.in`. And do we still need to keep around the `SocketServer` class in `src/aktualizr_secondary/socket_server.h`?

- removed the `socket_activation` reference from the doxygen file.

- Yes, that's actually a TCP server implementation that the IP secondary relies on.

**Pull request merged**